File: ios/ios/Base/BaseViewModel.swift
//
//  BaseViewModel.swift
//  ios
//
//  Created by Mustafa Ozhan on 25/12/2020.
//  Copyright © 2020 orgName. All rights reserved.
//

import Foundation

protocol BaseViewModel: ObservableObject {
    func startObserving()
    func stopObserving()
}

File: client/src/jsMain/kotlin/com/github/mustafaozhan/ccc/client/base/BaseViewModel.kt
/*
 * Copyright (c) 2021 Mustafa Ozhan. All rights reserved.
 */

package com.github.mustafaozhan.ccc.client.base

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.MainScope
import kotlinx.coroutines.cancelChildren

@Suppress("EmptyDefaultConstructor")
actual open class BaseViewModel actual constructor() {

    protected actual val clientScope: CoroutineScope = MainScope()

    protected actual open fun onCleared() {
        clientScope.coroutineContext.cancelChildren()
    }
}
